ocaml/xapi/storage_migrate.ml

```diff
@@ -369,7 +371,7 @@ let copy' ~task ~dbg ~sr ~vdi ~url ~dest ~dest_vdi =
 
     Pervasiveext.finally (fun () ->
         debug "activating RW datapath %s on remote" remote_dp;
-        ignore(Remote.VDI.attach2 dbg remote_dp dest dest_vdi true);
```

Comment:
It should be possible to find out the domid on the remote, passing an empty value here would likely break something for migration once we actually start using the domid in the plugin.

Reply:
Any ideas howto do this?

Reply:
Remote.VM.get_domid

Reply:
I think that the remote migration API should not be changed and it should keep calling `Remote.VDI.attach` and `Remote.VDI.attach2` (old versions of XenServer will call these anyway so the code needs to handle them), these are likely Dom0 attaches, so in the implementation of `attach` and `attach2` you would set domid to 0. 

We just need to make sure that whenever we attach something to a VM we set domid properly,
and that location is VBD plug/unplug.

In xenopsd for a VBD plug you would call `attach3` with the proper domid.

Reply:
done